Client/mods/deathmatch/logic/luadefs/CLuaElementDefs.cpp

```diff
@@ -953,43 +954,30 @@ int CLuaElementDefs::GetElementsWithinColShape(lua_State* luaVM)
     return 1;
 }
 
-int CLuaElementDefs::GetElementsWithinRange(lua_State* luaVM)
+CClientEntityResult CLuaElementDefs::GetElementsWithinRange(CVector pos, float radius, std::optional<std::string> type,
+    std::optional<unsigned short> interior, std::optional<unsigned short> dimension)
 {
-    CVector position;
-    float   radius;
-    SString elementType;
+    const auto typeHash = (type.has_value() && !type.value().empty()) ?
+        CClientEntity::GetTypeHashFromString(type.value()) : 0;
+        
+    CClientEntityResult result;
+    GetClientSpatialDatabase()->SphereQuery(result, CSphere{ pos, radius });
 
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadVector3D(position);
-    argStream.ReadNumber(radius);
-    argStream.ReadString(elementType, "");
-
-    if (!argStream.HasErrors())
-    {
-        // Query the spatial database
-        CClientEntityResult result;
-        GetClientSpatialDatabase()->SphereQuery(result, CSphere{position, radius});
+    // Remove elements that do not match the criterias
+    result.erase(std::remove_if(result.begin(), result.end(), [&](CClientEntity* pEntity) {
```

Comment:
This should only run if `typeHash != 0 || interior.has_value() || dimension.has_value()` is true

Reply:
Sure, I'll do that.